typing/typemod.ml

```diff
@@ -3100,7 +3100,8 @@ and type_structure ?(toplevel = None) funct_body anchor env sstr =
         Tstr_eval (expr, sort, attrs), [], shape_map, env
     | Pstr_value (rec_flag, sdefs) ->
         let (defs, newenv) =
-          Typecore.type_binding env rec_flag ~force_toplevel sdefs in
+          (* jra: should this be hard-coded? *)
+          Typecore.type_binding env Immutable rec_flag ~force_toplevel sdefs in
```

Comment:
This looks correct to me - you've ruled out the possibility that structure-level bindings are mutable in the parser.